### PR TITLE
refactor around components

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Omit
Docs: Omit

Some work previous to https://github.com/conan-io/conan/issues/13758

### CODE REVIEW

File: conan/tools/cmake/cmakedeps/templates/__init__.py [hunk 1]
```diff
@@ -30,20 +30,11 @@ def suffix(self):
             return ""
         return self.cmakedeps.build_context_suffix.get(self.conanfile.ref.name, "")
 
-    @property
-    def build_modules_activated(self):
```
Comment (memsharded) [inline conan/tools/cmake/cmakedeps/templates/__init__.py:34]: Not removed, just moved to the only place it is being used

File: conan/tools/cmake/cmakedeps/templates/target_data.py [hunk 2]
```diff
@@ -26,10 +26,17 @@ def filename(self):
         data_fname += "-data.cmake"
         return data_fname
 
+    @property
+    def _build_modules_activated(self):
```
Comment (memsharded) [inline conan/tools/cmake/cmakedeps/templates/target_data.py:30]: Moved from above, reducing scope of this method

File: conan/tools/cmake/cmakedeps/templates/target_data.py [hunk 3]
```diff
@@ -195,22 +202,22 @@ def _get_required_components_cpp(self):
                                                 self.is_host_windows)
 
             public_comp_deps = []
-            for require in comp.requires:
-                if "::" in require:  # Points to a component of a different package
-                    pkg, cmp_name = require.split("::")
```
Comment (memsharded) [inline conan/tools/cmake/cmakedeps/templates/target_data.py:200]: Trying to remove a bit the "parsing" of ``cpp_info.requires`` from the Generators. I have tried to go further in the abstraction, but failed.

File: conan/tools/microsoft/msbuilddeps.py [hunk 4]
```diff
@@ -319,21 +319,19 @@ def _package_props_files(self, require, dep, build=False):
         if dep.cpp_info.has_components:
             pkg_aggregated_content = None
             for comp_name, comp_info in dep.cpp_info.components.items():
-                if comp_name is None:
```
Comment (memsharded) [inline conan/tools/microsoft/msbuilddeps.py:322]: Component name can no longer be ``None`` that was already changed.

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**